pkg/flavor/flavor.go

```diff
@@ -0,0 +1,83 @@
+package flavor
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	virtv1 "kubevirt.io/client-go/api/v1"
+	flavorv1alpha1 "kubevirt.io/client-go/apis/flavor/v1alpha1"
+	flavorclient "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/flavor/v1alpha1"
+)
+
+var (
+	// These functions are exposed as variebles to allow mocking for unittests
+	FindProfile = findProfile
+	ApplyToVmi  = applyToVmi
+)
+
+func findProfile(vm *virtv1.VirtualMachine, client flavorclient.FlavorV1alpha1Interface) (*flavorv1alpha1.VirtualMachineFlavorProfile, error) {
+	if vm.Spec.Flavor == nil {
+		return nil, nil
+	}
+
+	isGlobalFlavor := vm.Spec.Flavor.GloballyScoped == nil || *vm.Spec.Flavor.GloballyScoped
```

Comment:
I agree, that it is counter-intuitive, but it was specified this way in the design document. The reasoning is that cluster flavors will be created by admin, and will be available to all VMs. And namespaced flavors can be created by users in their namespace for their VMs.

I think it may be cleaner to put the global flavors in a single namespace, like we do with common templates. But it would be more complex to implement.

Reply:
> This means global is the default, right?
> If so it feels counterintuitive to other parts of Kubernetes where the default is local/same namespace.

@kwiesmueller so, you're proposing that the vm flavor matcher defaults to assuming the flavor is namespace scoped within the vm's namespace? And then explicility opting into matching a globally scoped flavor?

The reason we leaned to having GloballyScoped default to `true` is because that's the anticipated primary use case. I'm not aware of any k8s precedence this conflicts with.

Reply:
> @kwiesmueller so, you're proposing that the vm flavor matcher defaults to assuming the flavor is namespace scoped within the vm's namespace? And then explicility opting into matching a globally scoped flavor?
> 
> The reason we leaned to having GloballyScoped default to `true` is because that's the anticipated primary use case. I'm not aware of any k8s precedence this conflicts with.  
> 

Right. Examples where references are handled like that could be pvc references in volumes, volumeClaimTemplates, etc.
It makes sense for this specific case to default this way, but i think it could be counter intuitive.

That aside, instead of the bool toggle, I think I'd still prefer specifying a Kind instead as cluster and namespaced are different kinds and defaulting that.

Reply:
lets get together and discuss this point a bit. I want to understand the value of using a Kind over a boolean to represent whether the object is globally scoped or not.

Reply:
@akrejcir Kevin and i caught up a bit about his suggestion here around using a `Kind` field to allow specifying either the cluster flavor or namespaced flavor.

I think going ahead with using `Kind` over the boolean would be the most future proof way of handling this. We can default `kind` to the cluster scoped flavor and people would only need to specify this when they want to use a namespaced flavor. 

what do you think?

Reply:
Ok I will change it to `kind`. Have you also discussed adding `group` field, or is `kind` enough?

Reply:
`kind` should be enough for now.

Reply:
Done